### PR TITLE
Add possibility to register multiple event sinks

### PR DESCRIPTION
**Pull request by glucaci**

### CODE REVIEW

File: src/IdentityServer/src/IdentityServer.Core/Events/IdOpsEventSink.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+using Microsoft.Extensions.Logging;
+
+namespace IdOps.IdentityServer.Events
+{
+    public class IdOpsEventSink : IEventSink
+    {
+        private readonly ILogger<BusEventSink> _logger;
+        private readonly IEnumerable<IIdOpsEventSink> _sinks;
+
+        public IdOpsEventSink(ILogger<BusEventSink> logger, IEnumerable<IIdOpsEventSink> sinks)
```
Comment (PascalSenn) [inline src/IdentityServer/src/IdentityServer.Core/Events/IdOpsEventSink.cs:16]:
```suggestion
        public IdOpsEventSink(
            ILogger<IdOpsEventSink> logger, 
            IEnumerable<IIdOpsEventSink> sinks)
```

File: src/IdentityServer/src/IdentityServer.Core/Events/IdOpsEventSink.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+using Duende.IdentityServer.Events;
+using Duende.IdentityServer.Services;
+using IdOps.IdentityServer.Extensions;
+using Microsoft.Extensions.Logging;
+
+namespace IdOps.IdentityServer.Events
+{
+    public class IdOpsEventSink : IEventSink
+    {
+        private readonly ILogger<BusEventSink> _logger;
```
Comment (PascalSenn) [inline src/IdentityServer/src/IdentityServer.Core/Events/IdOpsEventSink.cs:13]:
```suggestion
        private readonly ILogger<IdOpsEventSink> _logger;
```

File: src/IdentityServer/src/IdentityServer.Core/Events/IdOpsEventSink.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        {
+            _logger = logger;
+            _sinks = sinks;
+        }
+
+        public Task PersistAsync(Event evt)
+        {
+            try
+            {
+                Task.Run(() => ProcessAsync(evt)).Forget();
```
Comment (PascalSenn) [inline src/IdentityServer/src/IdentityServer.Core/Events/IdOpsEventSink.cs:26]: Forget?

File: src/IdentityServer/src/IdentityServer.Core/Events/IdOpsEventSink.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            {
+                _logger.LogError(ex, "Error while sending event");
+            }
+
+            return Task.CompletedTask;
+        }
+
+        private async Task ProcessAsync(Event evt)
+        {
+            foreach (IIdOpsEventSink sink in _sinks)
```
Comment (PascalSenn) [inline src/IdentityServer/src/IdentityServer.Core/Events/IdOpsEventSink.cs:38]: Do we want to process this in sequence or extract a method and do Task.WhenAll()?

File: src/IdentityServer/src/IdentityServer.Core/Events/IdOpsEventSink.cs (hunk 3 above)
Comment (glucaci) [reply src/IdentityServer/src/IdentityServer.Core/Events/IdOpsEventSink.cs:28]: It was like this in previous `BusEventSink`, seams that is fire and forget process

File: src/IdentityServer/src/IdentityServer.Core/Events/IdOpsEventSink.cs (hunk 4 above)
Comment (glucaci) [reply src/IdentityServer/src/IdentityServer.Core/Events/IdOpsEventSink.cs:40]: I think it's ok like this for the moment, we have to take a look maybe also to the forget action

**Pull request merged**